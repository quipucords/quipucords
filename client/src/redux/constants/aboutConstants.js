const ABOUT_DIALOG_OPEN = 'ABOUT_DIALOG_OPEN';
const ABOUT_DIALOG_CLOSE = 'ABOUT_DIALOG_CLOSE';

export { ABOUT_DIALOG_OPEN, ABOUT_DIALOG_CLOSE };
