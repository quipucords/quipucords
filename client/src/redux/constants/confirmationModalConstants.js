const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

export { CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
