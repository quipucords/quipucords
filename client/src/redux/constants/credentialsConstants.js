
const LOAD_CREDENTIALS_SUCCESS = 'LOAD_CREDENTIALS_SUCCESS';

export {
  LOAD_CREDENTIALS_SUCCESS
}
