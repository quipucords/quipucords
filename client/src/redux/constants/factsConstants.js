const ADD_FACTS_ERROR = 'ADD_FACTS_ERROR';
const ADD_FACTS_LOADING = 'ADD_FACTS_LOADING';
const ADD_FACTS_SUCCESS = 'ADD_FACTS_SUCCESS';

export { ADD_FACTS_ERROR, ADD_FACTS_LOADING, ADD_FACTS_SUCCESS };
