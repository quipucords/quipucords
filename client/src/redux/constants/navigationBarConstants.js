const NAV_TOGGLE_COLLAPSE = 'NAV_TOGGLE_COLLAPSE';
const NAV_COLLAPSE = 'NAV_COLLAPSE';
const NAV_OPEN = 'NAV_OPEN';

export { NAV_TOGGLE_COLLAPSE, NAV_COLLAPSE, NAV_OPEN };
