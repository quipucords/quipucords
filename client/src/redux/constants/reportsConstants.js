const GET_REPORT = 'GET_REPORT';
const GET_MERGE_REPORT = 'GET_MERGE_REPORT';

export { GET_REPORT, GET_MERGE_REPORT };
