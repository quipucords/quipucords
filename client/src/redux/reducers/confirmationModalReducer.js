import { confirmationModalTypes } from '../constants';

const initialState = {
  show: false,
  title: 'Confirm',
  heading: null,
  icon: null,
  body: null,
  confirmButtonText: 'Confirm',
  cancelButtonText: 'Cancel'
};

const confirmationModalReducer = (state = initialState, action) => {
  switch (action.type) {
    case confirmationModalTypes.CONFIRMATION_MODAL_SHOW:
      return Object.assign({}, state, {
        show: true,
        title: action.title,
        heading: action.heading,
        icon: action.icon,
        body: action.body,
        confirmButtonText: action.confirmButtonText || 'Confirm',
        cancelButtonText: action.cancelButtonText || 'Cancel',
        onConfirm: action.onConfirm,
        onCancel: action.onCancel
      });

    case confirmationModalTypes.CONFIRMATION_MODAL_HIDE:
      return Object.assign({}, state, {
        show: false
      });

    default:
      return state;
  }
};

confirmationModalReducer.initialState = initialState;

export { confirmationModalReducer as default, initialState, confirmationModalReducer };
