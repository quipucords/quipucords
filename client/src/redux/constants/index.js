import * as aboutTypes from './aboutConstants';
import * as credentialsTypes from './credentialsConstants';
import * as factsTypes from './factsConstants';
import * as navigationBarTypes from './navigationBarConstants';
import * as reportsTypes from './reportsConstants';
import * as scansTypes from './scansConstants';
import * as sourcesTypes from './sourcesConstants';
import * as viewToolbarTypes from './viewToolbarConstants';

const reduxTypes = {
  about: aboutTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  navigation: navigationBarTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes,
  aboutTypes,
  credentialsTypes,
  factsTypes,
  navigationBarTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  viewToolbarTypes
};

export default reduxTypes;
