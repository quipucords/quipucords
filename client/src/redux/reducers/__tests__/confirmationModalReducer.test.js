import confirmationModalReducer from '../confirmationModalReducer';
import { confirmationModalTypes as types } from '../../constants';

describe('ConfirmationModalReducer', () => {
  it('should return the initial state', () => {
    expect(confirmationModalReducer.initialState).toBeDefined();
  });

  it('should handle specific defined types', () => {
    const specificTypes = [types.CONFIRMATION_MODAL_HIDE, types.CONFIRMATION_MODAL_SHOW];

    specificTypes.forEach(value => {
      const dispatched = {
        type: value
      };

      const resultState = confirmationModalReducer(undefined, dispatched);

      expect({ type: value, result: resultState }).toMatchSnapshot(`defined type ${value}`);
    });
  });
});
