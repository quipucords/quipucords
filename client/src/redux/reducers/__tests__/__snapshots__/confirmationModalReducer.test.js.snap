// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defined type CONFIRMATION_MODAL_HIDE 1`] = `
Object {
  "result": Object {
    "body": null,
    "cancelButtonText": "Cancel",
    "confirmButtonText": "Confirm",
    "heading": null,
    "icon": null,
    "show": false,
    "title": "Confirm",
  },
  "type": "CONFIRMATION_MODAL_HIDE",
}
`;

exports[`defined type CONFIRMATION_MODAL_SHOW 1`] = `
Object {
  "result": Object {
    "body": undefined,
    "cancelButtonText": "Cancel",
    "confirmButtonText": "Confirm",
    "heading": undefined,
    "icon": undefined,
    "onCancel": undefined,
    "onConfirm": undefined,
    "show": true,
    "title": undefined,
  },
  "type": "CONFIRMATION_MODAL_SHOW",
}
`;
